x-pack/plugins/spaces/public/nav_control/nav_control_popover.tsx

```diff
@@ -154,6 +153,7 @@ export class NavControlPopover extends Component<Props, State> {
         aria-expanded={this.state.showSpaceSelector}
         aria-haspopup="true"
         aria-label={linkTitle}
+        data-test-subj="spacesNavSelector"
```

Comment:
I moved the test id to the button element of the popover rather than the popover itself. It looks like this solved the intermittent test failures in Firefox.

Reply:
Great idea. Last time I looked at this, I noticed the popover starts with `loading: false`, when you click it it flips to `loading: true` and then back to `loading: false`, so I'm not sure that really gave us the reliable behavior we desired anyway. I think this is a more solid approach.